src/controllers/submitUrlController.js

```diff
@@ -58,6 +65,52 @@ class SubmitUrlController extends UploadController {
   static urlIsNotTooLong (url) {
     return url.length <= 2048
   }
+
+  static async getHeadRequest (url) {
+    try {
+      return await axios.head(url)
+    } catch (err) {
+      console.error(err)
+      return null
+    }
+  }
+
+  static urlResponseIsNotTooLarge (response) {
+    try {
+      const contentLength = response.headers['content-length']
+
+      // Convert content length to MB
+      const sizeInMB = contentLength / (1024 * 1024)
+
+      return sizeInMB <= 10
+    } catch (err) {
+      console.error(err)
+      return true // for now we will allow this file as we can't be sure
+    }
+  }
+
+  static urlExists (response) {
+    if (!response) {
+      return false
+    }
```

Comment:
How come this is the only method that explicitly checks for this? Wouldn't you want this in all of them? Is it because this is the first one that is called of these new methods?

Reply:
if the url doesn't return anything which can happen with a non existing url. the response might be undefined. we could add it to the others. but as this one is called first. I don't think its needed in the others